elyra/kfp/operator.py

```diff
@@ -255,8 +263,15 @@ def __init__(
             self.container.set_memory_request(memory=str(mem_request) + "G")
 
         if self.gpu_limit:
-            gpu_vendor = self.pipeline_envs.get("GPU_VENDOR", "nvidia")
-            self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
+            if not self.gpu_vendor:
+                gpu_vendor = self.pipeline_envs.get("GPU_VENDOR", "nvidia")
+                self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
+            else:
+                self.container.add_resource_limit(self.gpu_vendor, self.gpu_limit)
+        if self.gpu_memory:
+            if not self.gpu_memory_vendor:
+                raise ValueError("gpu_memory_vendor not seted while gpu_memory is specified > 0")
```

Comment:
Typo
```suggestion
            if not self.gpu_memory_vendor:
                raise ValueError("gpu_memory_vendor not set while gpu_memory is specified")
```
Question:
This PR is making `gpu_memory_vendor` a required value.  Is there a reasonably common default (like "nvidia" is for `gpu_vendor`) that we can fall back to?  Are the same vendor names used for `gpu_limit` and `gpu_memory`?  I.e., can we use env `GPU_VENDOR` here as well if not provided by the user?

Reply:
"nvidia" I would consider to be the default due to the market share they hold. Not sure about the vendor spec when it comes to gpu limit and memory, I always assumed these would be agnostic?

Reply:
> Question:
> This PR is making gpu_memory_vendor a required value. Is there a reasonably common default (like "nvidia" is for gpu_vendor) that we can fall back to? Are the same vendor names used for gpu_limit and gpu_memory? I.e., can we use env GPU_VENDOR here as well if not provided by the user?

Thanks for reviewing!

Actually, `gpu_vendor`, `gpu_memory` and `gpu_memory_vendor` are ment to be optional. I'm looking for a way to make "nvidia" as default, as it is before, and make `gpu_vendor`, `gpu_memory` and `gpu_memory_vendor` to be optional.

Reply:
By the way, I've tested we can leave `gpu_vendor`, `gpu_memory` and `gpu_memory_vendor` blank, set `gpu` to 1, we'll have a step with default gpu resource setup like `nvidia.com/gpu: 1`.